bridge/client/app/_components/ktb-stage-overview/ktb-stage-overview.component.ts

```diff
@@ -1,77 +1,103 @@
+    withLatestFrom(this.project$),
+    filter(([stageName, project]) => Boolean(stageName && project)),
+    map(([stageName]) => stageName)
+  );
+
+  private readonly paramFilterType$ = this.route.queryParamMap.pipe(map((params) => params.get('filterType')));
+
+  public params$ = combineLatest([this.selectedStageName$, this.paramFilterType$, this.project$]).pipe(
+    tap(([stageName, filterType, project]) => {
+      const stage = project!.getStage(stageName!);
+      if (stage) {
+        this.selectedStageChange.emit({ stage: stage!, filterType: (filterType as ServiceFilterType) ?? undefined });
+      }
+    }),
+    takeUntil(this.unsubscribe$),
+    shareReplay(1)
+  );
+
   @Output() selectedStageChange: EventEmitter<{ stage: Stage; filterType: ServiceFilterType }> = new EventEmitter();
   @Output() filteredServicesChange: EventEmitter<string[]> = new EventEmitter<string[]>();
 
   constructor(private dataService: DataService, private apiService: ApiService, private route: ActivatedRoute) {}
 
-  public ngOnInit(): void {
-    // needs to be in init because of emitter
-    const project$ = this.route.params.pipe(
-      map((params) => params.projectName),
-      filter((projectName) => !!projectName),
-      tap(() => {
-        this.isTriggerSequenceOpen = this.dataService.isTriggerSequenceOpen;
-        this.dataService.isTriggerSequenceOpen = false;
-      }),
-      switchMap((projectName) => this.dataService.getProject(projectName)),
-      takeUntil(this.unsubscribe$)
-    );
-
-    project$.subscribe((project) => {
-      const differentProject = project?.projectName !== this.project?.projectName;
-      this.project = project;
-      this.setFilter(differentProject);
-    });
+  ngAfterViewInit(): void {
+    this.params$.subscribe();
   }
 
-  private setFilter(projectChanged: boolean): void {
+  private setFilter(project: Project | undefined, projectChanged: boolean): void {
     this._dataSource.data = {
       autocomplete: [
         {
           name: 'Services',
           autocomplete:
-            this.project?.getServices().map((service) => ({
+            project?.getServices().map((service) => ({
               name: service.serviceName,
             })) ?? [],
         } as DtAutoComplete,
       ],
     };
     this.globalFilter = this.apiService.environmentFilter;
     let newFilter: string[];
-    if (this.project) {
+    if (project) {
       // services can be deleted or added; adjust filter
-      const services = this.globalFilter[this.project.projectName]?.services || [];
+      const services = this.globalFilter[project.projectName]?.services || [];
       newFilter = services.filter((service) =>
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        this.project!.getServices().some((pService) => pService.serviceName === service)
+        project!.getServices().some((pService) => pService.serviceName === service)
       );
     } else {
       newFilter = [];
```

Comment:
Can be simplified to:
```
const createFilter = (project): string[] => {
  // services can be deleted or added; adjust filter
  const services = this.globalFilter[project.projectName]?.services || [];
  return services.filter((service) => project!.getServices().some((pService) => pService.serviceName === service);
} 
const newFilter: string[] = project ? createFilter(project) : [];
```

Reply:
We actually did not want to touch this method at all, but wanted to remove the `this.project` property. Should we still simplify the code?